dht_bootstrap.go

```diff
@@ -59,27 +61,54 @@ func (dht *IpfsDHT) startRefreshing() error {
 		}
 
 		for {
+			var waiting []chan<- error
 			select {
 			case <-refreshTicker.C:
-			case <-dht.triggerRtRefresh:
-				logger.Infof("triggering a refresh: RT has %d peers", dht.routingTable.Size())
+			case res := <-dht.triggerRtRefresh:
+				if res != nil {
+					waiting = append(waiting, res)
+				}
 			case <-ctx.Done():
 				return
 			}
-			dht.doRefresh(ctx)
+
+		collectWaiting:
+			for {
+				select {
+				case res := <-dht.triggerRtRefresh:
+					if res != nil {
+						waiting = append(waiting, res)
+					}
+				default:
+					break collectWaiting
+				}
+			}
+			err := dht.doRefresh(ctx)
+			for _, w := range waiting {
+				w <- err
```

Comment:
Close the channel?

Reply:
Not necessary.

Reply:
(well, if we're _returning_ a channel, we should probably close it)